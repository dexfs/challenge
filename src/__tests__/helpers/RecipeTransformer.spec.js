const RecipeTransformer = require('@app/helpers/RecipeTransformer')
const recipePuppyJson = require('../fixtures/recipepuppy.json')

describe('RecipeTransformer', () => {
  it('should transform data returned from service', () => {
    const transformer = new RecipeTransformer()
    expect(transformer.input(recipePuppyJson.results).get()).toMatchSnapshot()
  })
})
