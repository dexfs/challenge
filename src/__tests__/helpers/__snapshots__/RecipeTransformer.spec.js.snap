// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeTransformer should transform data returned from service 1`] = `
Array [
  Object {
    "ingredients": Array [
      "black pepper",
      "cream cheese",
      "eggs",
      "garlic",
      "italian seasoning",
      "milk",
      "olive oil",
      "onions",
      "parmesan cheese",
      "red pepper",
      "salt",
      "tomato",
      "vermicelli",
      "zucchini",
    ],
    "links": "http://find.myrecipes.com/recipes/recipefinder.dyn?action=displayRecipe&recipe_id=520763",
    "title": "Vegetable-Pasta Oven Omelet",
  },
  Object {
    "ingredients": Array [
      "bacon",
      "black pepper",
      "black pepper",
      "black pepper",
      "butter",
      "eggs",
      "garlic",
      "onions",
      "oregano",
      "roasted red peppers",
      "salt",
    ],
    "links": "http://www.bigoven.com/43919-Roasted-Pepper-and-Bacon-Omelet-recipe.html",
    "title": "Roasted Pepper and Bacon Omelet",
  },
  Object {
    "ingredients": Array [
      "basil",
      "broccoli",
      "eggs",
      "garlic",
      "lowfat milk",
      "onions",
      "parmesan cheese",
      "parmesan cheese",
      "salt",
      "tomato",
    ],
    "links": "http://cookeatshare.com/recipes/broccoli-oven-omelet-92851",
    "title": "
Broccoli Oven Omelet Recipe

",
  },
  Object {
    "ingredients": Array [
      "black pepper",
      "caraway seed",
      "coriander",
      "eggplant",
      "eggs",
      "garlic",
      "lemon",
      "olive oil",
      "onions",
      "salt",
    ],
    "links": "http://www.epicurious.com/recipes/food/views/Eggplant-Omelet-with-Coriander-and-Caraway-306",
    "title": "Eggplant Omelet with Coriander and Caraway",
  },
  Object {
    "ingredients": Array [
      "black pepper",
      "caraway seed",
      "coriander",
      "egg",
      "eggplant",
      "garlic",
      "lemon",
      "olive oil",
      "onions",
      "salt",
    ],
    "links": "http://www.bigoven.com/143831-Eggplant-Omelet-with-Coriander-And-Caraway-recipe.html",
    "title": "Eggplant Omelet with Coriander And Caraway",
  },
  Object {
    "ingredients": Array [
      "black pepper",
      "bread",
      "butter",
      "chicken broth",
      "cottage cheese",
      "egg substitute",
      "garlic",
      "mozzarella cheese",
      "onions",
      "oregano",
    ],
    "links": "http://www.bigoven.com/45151-Broccoli-and-Cheese-Omelet-recipe.html",
    "title": "Broccoli and Cheese Omelet",
  },
  Object {
    "ingredients": Array [
      "basil",
      "eggs",
      "garlic",
      "gruyere cheese",
      "leaves",
      "olive oil",
      "onions",
      "parmesan cheese",
      "parsley",
      "thyme",
    ],
    "links": "http://www.bigoven.com/149383-Chard-And-Onion-Omelet-recipe.html",
    "title": "Chard And Onion Omelet",
  },
  Object {
    "ingredients": Array [
      "eggs",
      "garlic",
      "olive oil",
      "onions",
      "parmesan cheese",
      "peas",
      "potato",
      "red pepper",
      "salt",
      "tomato",
      "zucchini",
    ],
    "links": "http://www.cdkitchen.com/recipes/recs/2184/Picnic-Omelet-Squares99498.shtml",
    "title": "Picnic Omelet Squares Recipe",
  },
  Object {
    "ingredients": Array [
      "black pepper",
      "chili powder",
      "eggs",
      "garlic",
      "green pepper",
      "milk",
      "mushroom",
      "onions",
      "salt",
      "vegetable oil",
      "water",
    ],
    "links": "http://www.cooks.com/rec/view/0,185,153160-249194,00.html",
    "title": "Spanish Omelet",
  },
  Object {
    "ingredients": Array [
      "almonds",
      "black pepper",
      "egg substitute",
      "eggs",
      "flour",
      "garlic",
      "goat cheese",
      "milk",
      "nonstick cooking spray",
      "olive oil",
      "onions",
      "parsley",
      "red wine vinegar",
      "salad greens",
      "salad greens",
      "salt",
      "thyme",
    ],
    "links": "http://find.myrecipes.com/recipes/recipefinder.dyn?action=displayRecipe&recipe_id=1622444",
    "title": "Onion and Fresh Herb Omelet with Mixed Greens",
  },
]
`;
